routes/v1.js

```diff
@@ -102,8 +116,83 @@ function addRoutes(app, peliasConfig) {
-    not(hasResponseDataOrRequestErrors), isPlaceholderServiceEnabled, isAdminOnlyAnalysis
+  const libpostalShouldExecute = all(
+    not(hasRequestErrors),
+    not(isRequestSourcesOnlyWhosOnFirst)
+  );
+
+  // execute placeholder if libpostal only parsed as admin-only and needs to
+  //  be geodisambiguated
+  const placeholderGeodisambiguationShouldExecute = all(
+    not(hasResponseDataOrRequestErrors),
+    isPlaceholderServiceEnabled,
+    // check request.clean for several conditions first
+    not(
+      any(
+        // layers only contains venue, address, or street
+        isOnlyNonAdminLayers,
+        // don't geodisambiguate if categories were requested
+        hasRequestCategories
+      )
+    ),
+    any(
+      // only geodisambiguate if libpostal returned only admin areas or libpostal was skipped
+      isAdminOnlyAnalysis,
+      isRequestSourcesOnlyWhosOnFirst
+    )
+  );
+
+  // execute placeholder if libpostal identified address parts but ids need to
+  //  be looked up for admin parts
+  const placeholderIdsLookupShouldExecute = all(
+    not(hasResponseDataOrRequestErrors),
+    isPlaceholderServiceEnabled,
+    // check clean.parsed_text for several conditions that must all be true
+    all(
+      // run placeholder if clean.parsed_text has 'street'
+      hasAnyParsedTextProperty('street'),
+      // don't run placeholder if there's a query or category
+      not(hasAnyParsedTextProperty('query', 'category')),
+      // run placeholder if there are any adminareas identified
+      hasAnyParsedTextProperty('neighbourhood', 'borough', 'city', 'county', 'state', 'country')
+    )
+  );
+
+  const searchWithIdsShouldExecute = all(
+    not(hasRequestErrors),
+    // don't search-with-ids if there's a query or category
+    not(hasAnyParsedTextProperty('query', 'category')),
+    // there must be a street
+    hasAnyParsedTextProperty('street')
+  );
+
+  // placeholder should have executed, useful for determining whether to actually
+  //  fallback or not (don't fallback to old search if the placeholder response
+  //  should be honored as is)
+  const placeholderShouldHaveExecuted = any(
+    placeholderGeodisambiguationShouldExecute,
+    placeholderIdsLookupShouldExecute
+  );
+
+  // don't execute the cascading fallback query IF placeholder should have executed
+  //  that way, if placeholder didn't return anything, don't try to find more things the old way
+  const fallbackQueryShouldExecute = all(
+    not(hasRequestErrors),
+    not(hasResponseData),
+    not(placeholderShouldHaveExecuted)
+  );
+
+  // defer to addressit for analysis IF there's no response AND placeholder should not have executed
+  const shouldDeferToAddressIt = all(
+    not(hasRequestErrors),
+    not(hasResponseData),
+    not(placeholderShouldHaveExecuted)
+  );
+
+  // call very old prod query if addressit was the parser
+  const oldProdQueryShouldExecute = all(
+    not(hasRequestErrors),
+    isAddressItParse
   );
 
```

Comment:
I think the predicate wrappers are fine as they are since they're in a state of flux and may end up being replaced by express functionality in the nearish future.  Having them all in routes/v1 makes referencing easier currently.